pod_container_marshalling.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright 2017 The go-marathon Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package marathon
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+// PodContainerAlias aliases the PodContainer struct so that it will be marshaled/unmarshaled automatically
+type PodContainerAlias PodContainer
+
+// UnmarshalJSON unmarshals the given PodContainer JSON as expected except for environment variables and secrets.
+// Environment variables are stored in the Env field. Secrets, including the environment variable part,
+// are stored in the Secrets field.
+func (p *PodContainer) UnmarshalJSON(b []byte) error {
+	aux := &struct {
+		*PodContainerAlias
+		Environment map[string]interface{} `json:"environment"`
+	}{
+		PodContainerAlias: (*PodContainerAlias)(p),
+	}
+	if err := json.Unmarshal(b, aux); err != nil {
+		return fmt.Errorf("malformed pod container definition %v", err)
+	}
+	env := map[string]string{}
+	secrets := map[string]Secret{}
+
+	for envName, genericEnvValue := range aux.Environment {
+		switch envValOrSecret := genericEnvValue.(type) {
+		case string:
+			env[envName] = envValOrSecret
+		case map[string]interface{}:
+			for secret, secretStore := range envValOrSecret {
+				if secStore, ok := secretStore.(string); ok && secret == "secret" {
+					secrets[secStore] = Secret{EnvVar: envName}
+					break
+				}
+				return fmt.Errorf("unexpected secret field %v or value type %T", secret, envValOrSecret[secret])
```

Comment:
I think `or` should say `of`.

The typo was already in the application equivalent. Could you please fix both?